### PR TITLE
Single video branch

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
- I've added single video page feature to my video library.
- Added history & liked feed

**ONLY check the following files**

#### Components: 
- [x] [components/VideoCard.jsx](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-1461aba6ad23131521bb371ae9c7965c273560004ea4e2d74a3f5efe2b57ee1e)

#### Pages files

- [x] [pages/HistoryFeed.js](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-7c5b2618630e80ce54b94373841fc99257ff05dc39a4043caff297a2ef5876a9)
- [x] [pages/LikedFeed.js](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-0fe227e6e04c750802cb51a79914e4695088230047b09357e7a3fe76bb33546f)
- [x] [pages/SingleVideoFeed.js](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-5140fd92ec91d9728026e6c752ef67bb91390440a27d1935a87ac2c65f5fa873)

 #### Contexts  + Reducers

- [x] [contexts/HistoryContext.js](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-bd13ade93621dd66671a96d03a7553f6a49ba89a5e4fd5a0c3b79e6bfc070fe6)
- [x] [contexts/LikesContext.js](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-648e5beafc1019eddaf086c1c3a54d6877b485fdb4167e3b99ca6a3359dc5a9e)
- [x] [reducers/historyReducer.js](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-cd6dc0506bc00bfcc6b4fb88585c094d8d5ff2db25f2dccc52089a2acc45e458)
- [x] [reducers/likesReducer.js](https://github.com/nayyyhaa/inNotion-watch/compare/single-video-branch?expand=1#diff-7195c071e89b7c7438c6a589fe6b54443d917f70d5dd95413325bb941bdff1b6)

#### files to *ignore*
- [ ] data files
- [ ] CSS files
- [ ] index files

### CODE REVIEW

File: src/reducers/historyReducer.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const historyReducer = (state, action) => {
+  switch (action.type) {
+    case "ADD_TO_HISTORY":
+      return [...new Set([...state, action.payload])];
```
Comment (foolhardy21) [inline src/reducers/historyReducer.js:4]: good use of `Set` for adding unique videos to history.

File: src/pages/HistoryFeed.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import noVideo from "toolkit/assets/search.svg";
+
+export const HistoryFeed = () => {
+  const { history } = useHistory();
+  return (
+    <>
+      <h1 className="title colored-text centered-text m-t-3">History</h1>
+      <div className="full-wd m-auto row-flex align-start p-v-2 p-h-5 m-t-3">
+        {history.length > 0 ? (
+          history?.map((video) => <VideoCard key={video._id} video={video} cardData="history" />)
```
Comment (foolhardy21) [inline src/pages/HistoryFeed.js:12]: good reuse of `VideoCard` based on prop `cardData`

**Review by foolhardy21 — APPROVED** [review]
Nice work. the code and logic used seems good to me. I have mentioned some things that I liked. Good to go ahead.

**Review by ApurvaSawant11 — APPROVED** [review]
Everything looks great! Did not find anything to suggest. Good work

**Comment by nayyyhaa** [comment]
Thanks @ApurvaSawant11 & @foolhardy21 for reviewing. merging the PR

**Pull request merged**